axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/PrimaryEventStore.java

```diff
@@ -329,7 +334,8 @@ private void completeSegment(WritePosition writePosition) {
         }
     }
 
-    private void write(WritePosition writePosition, int eventSize, List<ProcessedEvent> eventList) {
+    private void write(WritePosition writePosition, int eventSize, List<ProcessedEvent> eventList,
```

Comment:
maybe i.o. an input/output argument can be used the return value?

Reply:
Decided not to do this, because there is a (minimal) risk of a concurrency issue, that the synchronizer completes the block before the write returned.